### DISCUSSION TITLE
Github code search is showing stale results

### DISCUSSION THREAD

**Discussion by vivekweb2013** [community/community#13516] [Other Feature Feedback, Questions, & Ideas] [answered]
Github code search is not real-time. I had to search for 3 or 4 time sometimes for the search result to appear.
One can easily reproduce this by creating a new file in the repository and searching immediately after pushing your changes.
It will take around a minute or so to show the newly introduced changes on the search result page.

Same is the case with deleting the files and searching. The search result page will show the deleted files in the result page and shows a 404 page when user tries to access the file.

**Comment by colinwm** [comment, marked as answer]
You're right, there is a several minute delay between merging a PR and being able to search for those changes.

**Reply by Markus-Bernt** [reply]
Has the delay been extended? 

I added a test file for a test search over 24 hours ago and I also removed four files that where found through the test search. 
Both changes are still not visible in the search results. => Instead, it still shows the results of a two-month old commit.

---

My question relates to the GitHub API endpoint: `/search/code` 
=> Here is my [original question](https://github.community/t/github-api-code-search-outdated-search-results/262590) with detailed problem description and an example.

**Reply by colinwm** [reply]
Ah, my answser is in relation to the new code search experience at cs.github.com. For the API, the delay may be much longer.

**Reply by Markus-Bernt** [reply]
Thank you for the quick reply! :)

- Do you may know where i can see the delay time of the API? Because i think a delay of over 48 hours is clearly too long.. 
- Or do you know if there is a way to speed up the indexing?

_(I apologize if this is the wrong discussion area for that.)_

**Reply by ronenfe** [reply]
That's not good, you don't want a sensitive data to show in the search results even after you deleted the repository. Why not run the search on the current data?

**Reply by look** [reply]
@ronenfe At the time this was answered, the code search API used a different indexing system than the web. Since then, it has been switched to the new system and the timeline for updating content should be the same (a few minutes in most cases).
